pkg/internal/cryptoutil/utils.go

```diff
@@ -50,20 +69,78 @@ func IsChachaKeyValid(key []byte) bool {
 // KeyPair represents a private/public key pair
 type KeyPair struct {
 	// Priv is a private key
-	Priv []byte
+	Priv []byte `json:"priv,omitempty"`
 	// Pub is a public key
-	Pub []byte
+	Pub []byte `json:"pub,omitempty"`
 }
 
-// IsKeyPairValid is a utility function that validates a KeyPair
-func IsKeyPairValid(kp KeyPair) bool {
+// EncKeyPair represents a private/public encryption key pair
+type EncKeyPair struct {
+	KeyPair `json:"keypair,omitempty"`
+	// Alg is the encryption algorithm of keys enclosed in this key pair
+	Alg EncryptionAlgorithm `json:"alg,omitempty"`
+}
+
+// SigKeyPair represents a private/public signature (verification) key pair
+type SigKeyPair struct {
+	KeyPair `json:"keypair,omitempty"`
+	// Alg is the signature algorithm of keys enclosed in this key pair
+	Alg SignatureAlgorithm `json:"alg,omitempty"`
+}
+
+// KeyPairCombo represents a pair of key pairs, one for encryption and one for signature
```

Comment:
Combo doesn't have the right ring to it.

Reply:
I will rename to `MessagingKeys`